modules/DragonridingVigor.lua

```diff
@@ -10,14 +17,107 @@ local DragonridingBuffs = {
 	412088, -- Grotto Netherwing Drake
 }
 
+-- Old Vigor widget (pre-11.2.7)
 local vigorWidgetSetID = 283
 local vigorWidgetType = 24
 local defaultVigorWidgetID = 4460
 local vigorWidgetIDs = nil
+
+-- New Skyriding charges (11.2.7+)
+-- Verified IDs:
+--   Surge Forward (Skyriding mount ability): 372608
+--   Skyward Ascent (Skyriding mount ability): 372610
+local SKYRIDING_CHARGE_SPELL_IDS = { 372608, 372610 }
+
 local knowsAlternateMountEnum = Enum and Enum.PowerType and Enum.PowerType.AlternateMount
 local unitPowerType = Enum and Enum.PowerType and Enum.PowerType.AlternateMount
 unitPowerType = unitPowerType or ALTERNATE_POWER_INDEX
 
+
+local BUILD_NUMBER = select(4, GetBuildInfo())
+local FORCE_CHARGES_BUILD = 110207 -- 11.2.7
```

Comment:
Is there a way we can feature-flag this rather than checking a build number? I prefer to conditionalize on the availability of a new spell/API/feature rather than a build number since Blizzard has backported engine changes many times and build-number checking tends to cause Classic clients to start behaving weirdly when they do that.

In this instance, it doesn't seem likely that Dragonriding changes would need to work in a Classic environment, but if we can feature-flag for the sake of consistency, that's the most robust method.

Reply:
It looks like `UsingOldVigorWidgets()` may be the appropriate feature-flagged way to do this, and we might not need the build number check at all?